wifiphisher/common/interfaces.py

```diff
@@ -366,13 +451,18 @@ def is_interface_valid(self, interface_name, mode=None):
         # raise an error if interface can't be found
         try:
             interface_adapter = self._name_to_object[interface_name]
+            if mode == "internet":
+                interface_adapter.is_internet_access = True
 
             # raise an error if interface doesn't support the mode
+            if mode != "internet" and interface_adapter.is_managed_by_nm:
```

Comment:
I think the one that I showed you would be better where we check if it's not `internet` and not managed by the network manager. Because here you are not checking if the `monitor` or the `AP` interfaces are managed.